polynote-frontend/polynote/cell.js

```diff
@@ -448,30 +510,16 @@ export class TextCell extends Cell {
                 i++;
             }
 
-            // the edits have to be applied in order - pos is stateful wrt the edits that have been seen.
-            // we'll take at most one deletion and one addition as a single edit.
-            let text = "";
-            let len = 0;
             if (i < diff.length) {
-                if (diff[i].added) {
-                    while (i < diff.length && diff[i].added) {
-                        text = text + diff[i].value;
-                        i++;
-                    }
-                    edits.push(new messages.ContentEdit(pos, 0, text));
-                    pos += text.length;
-                } else if (diff[i].removed) {
-                    while (i < diff.length && diff[i].removed) {
-                        len += diff[i].value.length;
-                        i++;
-                    }
-                    while (i < diff.length && diff[i].added) {
-                        text = text + diff[i].value;
-                        i++;
-                    }
-                    edits.push(new messages.ContentEdit(pos, len, text));
+                const d = diff[i];
+                const text = d.value;
+                if (d.added) {
```

Comment:
Is it guaranteed that `d.added !== d.removed`? If so, it seems like it would be wayy clearer if we used an enum for this rather than relying on two flags (I know that diff.js is lifted from some other project but still)

Reply:
I think it is guaranteed that `added && removed == false`. But TBH I'm not totally sure. Agree that a different mechanism would make more sense, but I think JsDiff code will go away once we get rid of the hand-rolled markdown editor and move to ProseMirror. So I'm not that worried about it.